keras_cv/models/object_detection/mask_rcnn/mask_rcnn.py

```diff
@@ -0,0 +1,307 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Mask R-CNN model based on the Tensorflow model garden implementation.
+
+As an experiment, Keras CV will try to wrap a thin layer on top of model garden
+implementation, so that we could quickly provide models to end user without
+re-implement/re-train the model end to end.
+
+This class might serve as a temporary solution, and might change its interface
+and details if needed in the future.
+"""
+
+import tensorflow as tf
+
+try:
+    import tensorflow_models as tfm
+except ImportError:
+    # Didn't have tensorflow model garden package installed, and the model can't
+    # be instantiated.
+    tfm = None
+
+
+def _assert_tensorflow_model_installed():
+    if tfm is None:
+        raise ImportError(
+            'Tensor model package is not installed in the '
+            'environment. Please use `pip install tf-models-official` '
+            'or `pip install tf-models-nightly` to install the required '
+            'dependency package')
+
+
+def mask_rcnn(
```

Comment:
is `mask_rcnn` our intended API surface?

I think that by wrapping in a sub-model similar to the RetinaNet we will be able to better manage the bounding_box_format, include_rescaling, and metric evaluation during fit().

I think you'll also need to do this to get label encoding and prediction decoding working w/ model.fit(), model.evaluate(), and model.predict().

Reply:
yea, I think we probably should take the delegate approach, that make MaskRCNN to include a tfm MaskRCNN model, and delegate all the public methods to it.

My original thought was to directly subclass the tfm MaskRCNN model, but then this requires a hard dependency to tfm, otherwise the code will fail at the package load time when tfm is not installed.